extensions/rich_text_components/rte-output-display.component.spec.ts

```diff
@@ -110,4 +110,37 @@ describe('RTE display component', () => {
       component.ngAfterViewInit();
     }).toThrowError();
   }));
+
+  it('should not display type 3 nodes', () => {
+    const removeChildSpy = jasmine.createSpy('Remove child node');
+
+    component.elementRef = {
+      nativeElement: {
+        childNodes: [
+          {
+            nodeType: 3,
+            parentElement: {
+              removeChild: removeChildSpy
+            }
+          }
+        ]
+      }
+    };
```

Comment:
[Optional] I think if you use fixture.detectChanges(), you will not need to mock element ref. Also add a testcase for the problematic ones. One such problem we found was texts not wrapped in p tags. So one value of rte string can be `Congratulations you have finished`. Maybe @Nik-09 can add the test case he found for which the text was leaking.

Reply:
Done